integration/test/organization/main_test.go

```diff
@@ -0,0 +1,137 @@
+// +build k8srequired
+
+package organization
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/core/v1alpha1"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/crd"
+	"github.com/giantswarm/apptest"
+	"github.com/giantswarm/micrologger"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-admission-controller/integration/env"
+)
+
+const (
+	testCatalogName = "control-plane-test"
+	prodCatalogName = "control-plane"
+	testCatalogUrl  = "https://giantswarm.github.io/control-plane-test-catalog"
+	prodCatalogUrl  = "https://giantswarm.github.io/control-plane-catalog"
+	// API Groups for upstream Cluster API types.
+	clusterAPIGroup                    = "cluster.x-k8s.io"
+	infrastructureAPIGroup             = "infrastructure.cluster.x-k8s.io"
+	experimentalClusterAPIGroup        = "exp.cluster.x-k8s.io"
+	experimentalInfrastructureAPIGroup = "exp.infrastructure.cluster.x-k8s.io"
+)
+
+var (
+	appTest apptest.Interface
+	logger  micrologger.Logger
+)
+
+// TestMain allows us to have common setup and teardown steps that are run
+// once for all the tests https://golang.org/pkg/testing/#hdr-Main.
+func TestMain(m *testing.M) {
+	var err error
+
+	ctx := context.Background()
+
+	{
+		logger, err = micrologger.New(micrologger.Config{})
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		appTest, err = apptest.New(apptest.Config{
+			KubeConfigPath: env.KubeConfig(),
+			Logger:         logger,
+		})
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		values := `Installation:
```

Comment:
If this can be done in a lightweight way I think this would be nice.

My only concern is from past experience with the e2etemplates library. Some of those templates got complex and needing to update multiple repos slows down working on tests.

Reply:
Yeah, I understand that, but here it would be in the same repo so that wouldn't be so much of a problem. I've been recently using a lot of pattern where unit test input CRs are stored as file and loaded during the test and I've personally liked that a lot. Removes a bit of boilerplate from the test code. See e.g. https://github.com/giantswarm/azure-operator/blob/master/integration/test/crmapping/crmapping_test.go#L110-L177

Reply:
I tried the template thingy for the values. Take a look @tuommaki